### PR TITLE
[refactor][build] Collapse empty tags and duplicated dependences in pom.xml

### PR DESCRIPTION
**Pull request by presenth7s**
### Modifications

- Dependency is duplicated in file(s): Pulsar、Pulsar-io/netty、structured-event-log.  And there are already duplicate dependencies in the root file pom.xml, open source considers using maven dependency delivery to solve the problem of duplicate dependencies

`<dependency>`
       `<groupId>org.testng</groupId>`
       `<artifactId>testng</artifactId>`
       `<scope>test</scope>`
` </dependency>`

- XML: collapse empty tags

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [x] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Comment by presenth7s** [comment]
@momo-jun tips: First-time contributors need a maintainer to approve running workflows.  And Can the workflow be approved according to the specification ?  ths

**Comment by momo-jun** [comment]
@presenth7s thanks for your contribution. 
As mentioned in the PR template, changes to code are required to run the workflow for testing in the local forked repo first to save GitHub Actions' constrained resources and quota. Can you pls attach the test PR in your forked repo?

**Comment by presenth7s** [comment]
> @presenth7s thanks for your contribution. As mentioned in the PR template, changes to code are required to run the workflow for testing in the local forked repo first to save GitHub Actions' constrained resources and quota. Can you pls attach the test PR in your forked repo?

Well, this is the first time that I need to execute CI in my own repo first, and do I need to resubmit the PR?

**Comment by presenth7s** [comment]
@momo-jun Well, this is the first time that I need to execute CI in my own repo first, and do I need to resubmit the PR?

**Comment by tisonkun** [comment]
> this is the first time that I need to execute CI in my own repo first

@presenth7s You can read the contribution guide (https://pulsar.apache.org/contribute/personal-ci) for the background and instructions. I'm reviewing this change now :)

**Review by tisonkun — APPROVED** [review]
LGTM.

**Comment by tisonkun** [comment]
> seems to be stuck on "Pulsar CI checks completed "

What's the exact issue you meet? Where is your PR in the forked repo?

File: pulsar-io/netty/pom.xml [hunk 1]
```diff
@@ -68,13 +68,6 @@
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-lang3</artifactId>
         </dependency>
-
-        <dependency>
```
Comment (nodece) [inline pulsar-io/netty/pom.xml:72]: I notice that the pulsar-io-netty requires the TestNG. Why remove this dependency?

Comment (presenth7s) [reply pulsar-io/netty/pom.xml:72]: Dependency is duplicated in file(s): Pulsar、Pulsar-io/netty、structured-event-log. And there are already duplicate dependencies in the root file pom.xml, open source considers using maven dependency delivery to solve the problem of duplicate dependencies

Comment (tisonkun) [reply pulsar-io/netty/pom.xml:72]: @nodece test scope TestNG is globally depended in the root `pom.xml`.

**Review by nodece — APPROVED** [review]

**Comment by presenth7s** [comment]
@momo-jun @tisonkun  @nodece  Pulsar CI checks completed.  thank you for the help

**Comment by presenth7s** [comment]
@momo-jun @tisonkun @nodece  Pulsar CI checks completed. And thank you for the help

**Comment by tisonkun** [comment]
@presenth7s the first run has several failures and I retrigger the tests for you. We're suffering a number of flaky tests. You can filter issues by the [flaky-tests](https://github.com/apache/pulsar/issues?q=is%3Aopen+is%3Aissue+label%3Aflaky-tests) label.

**Comment by tisonkun** [comment]
Merging...

@presenth7s Thanks for your contribution!

**Pull request merged**